aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1196,10 +1196,8 @@ Tensor& orgqr_out(const Tensor& input, const Tensor& tau, Tensor& result) {
   //   "result shape ", result.sizes(), " does not match input shape ", input.sizes());
   // }
 
-  // Single matrix MAGMA routine requires 'infos' to reside in CPU memory,
-  // therefore we create 'infos' only on CPU for now.
-  // This should be changed if cuSOLVER would be used
-  auto infos = at::empty({std::max<int64_t>(1, batchCount(input))}, input.options().dtype(kInt).device(kCPU));
+  // cuSOLVER is used for CUDA inputs and it requires 'infos' to reside in GPU memory
```

Comment:
I thought either MAGMA or cuSOLVER could be used?

Reply:
I should make this comment more clear. Yes, both are used. cuSOLVER requires `infos` to be on GPU, MAGMA path doesn't use it (it can't be used).